tests/hopp/test_wave.py

```diff
@@ -0,0 +1,121 @@
+				[6.250000, 0.000000, 0.000000, 0.000000, 0.000000, 219.200000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 244.200000, 201.200000, 166.400000, 138.200000, 115.500000, 97.100000, 0.000000],
+				[6.750000, 0.000000, 0.000000, 0.000000, 0.000000, 255.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 281.900000, 232.400000, 192.200000, 159.700000, 133.500000, 112.300000, 0.000000],
+				[7.250000, 0.000000, 0.000000, 0.000000, 0.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 265.600000, 219.800000, 182.800000, 152.900000, 128.700000, 0.000000],
+				[7.750000, 0.000000, 0.000000, 0.000000, 0.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 249.000000, 207.200000, 173.400000, 146.000000, 0.000000],
+				[8.250000, 0.000000, 0.000000, 0.000000, 0.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 279.900000, 233.000000, 195.100000, 164.400000, 0.000000],
+				[8.750000, 0.000000, 0.000000, 0.000000, 0.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 260.200000, 218.000000, 183.800000, 0.000000],
+				[9.250000, 0.000000, 0.000000, 0.000000, 0.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 242.100000, 204.100000, 0.000000],
+				[9.750000, 0.000000, 0.000000, 0.000000, 0.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 286.000000, 267.400000, 225.600000, 0.000000]
+			]
+        }
+
+cost_model_inputs = {
+	'reference_model_num':3,
+	'water_depth': 100,
+	'distance_to_shore': 80,
+    'number_rows': 10,
+	'devices_per_row':10,
+	'device_spacing':600,
+	'row_spacing': 600,
+	'cable_system_overbuild': 20
+}
+
+
+def test_changing_n_devices():
+    # test with gridded layout
+    model = MHKWavePlant(site = site, mhk_config=mhk_config,cost_model_inputs=cost_model_inputs)
+    assert(model.system_capacity_kw == 28600)
+    for n in range(1, 20):
+        model.number_devices = n
+        assert model.number_devices == n, "n turbs should be " + str(n)
+        assert model.system_capacity_kw == pytest.approx(28600, 1), "system capacity different when n turbs " + str(n)
```

Comment:
since the cost model is built in, I would also test that the costs have been updated

Reply:
Added in https://github.com/NREL/HOPP/pull/133/commits/d0664456e19fe9440230514fe950ad3f9266344d